### PR TITLE
 donating to a project

### PR DESCRIPTION
**Pull request by Humancyyborg**
#endpoint that require  donor login before donating to a project
Screen Shot 2022-04-19 at 9 31 36 PM

@cyrushhc  Here you  go

### CODE REVIEW

**Comment by pabloguarneros** [comment]
Hi @Humancyyborg, thanks for writing the route.

I have a few questions before you commit.

1. I’ve seen on line 221 that you commented 

```python
#would have to look this up and complete the rest
```

What do you want to look up? It might be helpful to complete the rest before we merge.

2.  Could you add test cases to this endpoint? For example, what happens if the id of the project does not exist? Would we return the response in line 239 

```python
return { "response": 500 }
```

or would the code stop before executing this line?

3. The other routing functions are written in snake case. Could we change donateProject to snake case too? Using the same naming convention for Python would help us keep the code cleaner.

4. What’s the intention for this if-else statement?

``` python
if raise_amount==0:
	project.raise_amount = raise_amount
else:
	project.raise_amount+=raise_amount
```

It seems that if someone were to donate 0 dollars to the project and the project had raised 140 dollars so far, the project’s raised_amount would get reset. If the intention is to keep the project raise amount the same, would project.raise_amount+=raise_amount still do the job? If so, is the if-else statement necessary?

**Comment by Humancyyborg** [comment]
@pabloguarneros  sorry that was before I wrote the method. I was considering the best way to handle it.. sorry I left it there, I should have removed it. As for the requested changes, I will update that later tonight(maybe after capstone tonight [🥺])

**Comment by Mer-lone** [comment]
@Humancyyborg Echoing what @pabloguarneros said, consider deleting the commented section if not in use. From lines 138 to 140.

**Comment by Humancyyborg** [comment]
@Mer-lone  addressing the concerns today...actually now

**Comment by Humancyyborg** [comment]
> 

Hahaha..for concern 4... crazy folks might want to donate 0 dollars.  I am guessing throwing an exception and asking the user to enter a valid figure makes more sense from a dev point of view

**Comment by Humancyyborg** [comment]
@pabloguarneros  could elaborate on the issue you raised in number two. By test cases ,are you talking about test cases or doing an if-else check to see if the ID exists?

**Comment by Humancyyborg** [comment]
@pabloguarneros  @Mer-lone  applied some of the changes....Let me know if there is anything I need to add

**Comment by cyrushhc** [comment]
Hi @Humancyyborg, I looked into the conflict between this branch and the master, which is about the register() in route.py.   
I decided to use the master branch version as it was added in the latest master version by @Phil-Barv.

Let me know if there is issue with that.

**Comment by Phil-Barv** [comment]
@cyrushhc I have concerns that the changes made to routes.py here will conflict with the tests that have been written. For example, the changes to the return statements from returning a "200" status to "registered".

**Comment by cyrushhc** [comment]
Could you share more? Was the original version in this branch intended for testing?

**Comment by Phil-Barv** [comment]
@cyrushhc let me generate a PR, you'll see what I mean. Just a moment.

**Comment by cyrushhc** [comment]
@Phil-Barv Could you explain which specific part will conflict to test version? I don't see difference between this version of register() and the test version

File: backend/routes.py [hunk 1]
```diff
@@ -280,8 +280,6 @@ def register():
 
     return { "registered": registered }
```
Comment (Phil-Barv) [inline backend/routes.py:295]: @cyrushhc this is an example

Comment (Phil-Barv) [reply backend/routes.py:295]: But if you've checked and there are no differences then I think we're good.

Comment (cyrushhc) [reply backend/routes.py:295]: But that is what it looks like on the test-route branch. Could you double-check it?

Comment (Phil-Barv) [reply backend/routes.py:295]: @cyrushhc You are right!!! 'registered' is a Boolean variable. In the tests, it checks for the truth value (True/False). This is where the confusion came from.

**Comment by pabloguarneros** [comment]
> @pabloguarneros could elaborate on the issue you raised in number two. By test cases ,are you talking about test cases or doing an if-else check to see if the ID exists?

I meant a test case like @Phil-Barv did for the route functions.

``` python
class appClientTests(unittest.TestCase):

    def test_api_returns_data_packet(self):
        """
        Test that checks the /api route returns the exact json packet.
        """
        response = requests.get(f'{app}/api')
        expected_response = {
                                "tutorial": "Flask React Heroku"
                            }
    
        self.assertEqual(response.json(), expected_response)
```

**Comment by Humancyyborg** [comment]
@pabloguarneros  gotcha... I will handle that,no problem!

**Pull request merged**